### PR TITLE
Convenience method for product search, and snippets/docs

### PR DESCRIPTION
**Pull request by jskeet**
For the snippets to work, a product set must first be populated.
This commit includes a tool to create the product set, with
documentation for the tool described in TESTING.md.

### CODE REVIEW

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.Snippets/VisionFixture.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using Google.Cloud.ClientTesting;
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.Snippets/VisionFixture.cs]: This one is missing the copyright notice and Travis is complaining.

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.PopulateProductSearchTestSet/Program.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length != 2)
+            {
+                // FIXME
+                return;
+            }
+            string directory = args[0];
+            var json = File.ReadAllText(Path.Combine(directory, "products.json"));
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.PopulateProductSearchTestSet/Program.cs]: Just confirming: Did we already had the `products.json` file?

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.PopulateProductSearchTestSet/Program.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Google.Cloud.Vision.V1.PopulateProductSearchTestSet
+{
+    /// <summary>
+    /// Application to populate a sample set of products for tests and snippets.
+    /// TODO: Handle the set (and maybe the bucket) already existing.
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.PopulateProductSearchTestSet/Program.cs]: And when this TODO is done, maybe move this code to the test projects themselves, in a fixture or test init of some kind?

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.PopulateProductSearchTestSet/Program.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    /// </summary>
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length != 2)
+            {
+                // FIXME
```
Comment (jskeet) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.PopulateProductSearchTestSet/Program.cs]: Oops - I've just seen this FIXME after merging. Will fix it in a separate PR :)

**Pull request merged**